src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -36,4 +84,73 @@ protected function getExtension()
     {
         return 'yml';
     }
+
+    private function expandToTree(array $messages)
+    {
+        $tree = array();
+
+        foreach ($messages as $id => $value) {
+            $this->expandElement($tree, explode('.', $id), $value);
```

Comment:
how to deal with an id like [You must select at most {{ limit }} choice.|You must select at most {{ limit }} choices.](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Validator/Resources/translations/validators.en.xlf#L30)